libproto/src/lib.rs

```diff
@@ -234,20 +234,107 @@ impl Block {
         self.get_body().transactions_root().0 == *self.get_header().get_transactions_root()
     }
 
-    pub fn block_verify_req(&self, request_id: u64) -> VerifyBlockReq {
-        let mut reqs: Vec<VerifyTxReq> = Vec::new();
-        let signed_txs = self.get_body().get_transactions();
-        for signed_tx in signed_txs {
-            let signer = signed_tx.get_signer();
-            let unverified_tx = signed_tx.get_transaction_with_sig();
-            let mut verify_tx_req = unverified_tx.tx_verify_req_msg();
-            verify_tx_req.set_signer(signer.to_vec());
-            reqs.push(verify_tx_req);
+    pub fn compact(&self) -> CompactBlock {
```

Comment:
I suggest that rename to `compact_block`.

Reply:
In the most scenario, we would consume the block?

Reply:
1. Consume `self` is OK, I will update for all.
1. IMHO, the method named `compact()` like a `generic method`.